src/components/Task/Task.jsx

```diff
@@ -0,0 +1,26 @@
+import CloseIcon from "@mui/icons-material/Close";
+import { Div, P, Input, Button, PMark } from "./Task.styles.js";
+import { useState } from "react";
+
+export const Task = ({ text }) => {
+  const [markTask, setMarkTask] = useState(false);
+
+  const handleCheckboxChange = () => {
+    setMarkTask((prevMarkTask) => !prevMarkTask);
+  };
+
+  return (
+    <Div>
+      <Input
+        type="checkbox"
+        checked={markTask}
+        onChange={handleCheckboxChange}
+      />
+      {!markTask ? <P>{text}</P> : <PMark>{text}</PMark>}
```

Comment:
1. As often as possible you'd wanna not negate statements in conditions. More natural here would be "if task is marked..."
2. `P` and `PMark` are almost identical styles. Only color and `line-through` are applied here when the task is marked.

What you could do, is have only one component, that is parametrized. So what I'd wanna see here is sth like:
```
<P isMarked={isMarked}>{text}</P>
```

Here's a bit more on how to pass properties to Styled component: https://styled-components.com/docs/basics#passed-props

Once you learn this, you'll discover the beauty of Styled Components, b/c you'll notice, you can change styles on the fly with use of JavaScript. Hence the name of the approach: CSS-in-JS